internal/cutil/cslice.go

```diff
@@ -0,0 +1,92 @@
+package cutil
+
+// #include <stdlib.h>
+import "C"
+import "unsafe"
+
+// The following code needs some explanation:
+// This creates slices on top of the C memory buffers allocated before in
+// order to safely and comfortably use them as arrays. First the void pointer
+// is cast to a pointer to an array of the type that will be stored in the
+// array. Because the size of an array is a constant, but the real array size
+// is dynamic, we just use the biggest possible index value maxIdx, to make
+// sure it's always big enough. (Nothing is allocated by casting, so the size
+// can be arbitrarily big.) So, if the array should store items of myType, the
+// cast would be (*[maxIdx]myItem)(myCMemPtr).
+// From that array pointer a slice is created with the [start:end:capacity]
+// syntax. The capacity must be set explicitly here, because by default it
+// would be set to the size of the original array, which is maxIdx, which
+// doesn't reflect reality in this case. This results in definitions like:
+// cSlice := (*[maxIdx]myItem)(myCMemPtr)[:numOfItems:numOfItems]
+
+const maxIdx = 1<<31 - 1 // 2GB, max int32 value, should be safe
+const ptrSize = C.size_t(unsafe.Sizeof(unsafe.Pointer(nil)))
+
+// CPtr is a pointer to C memory.
+// Not required, but makes it more obvious, when only pointers to C memory and
+// no pointers to Go memory are allowed to be stored.
+type CPtr unsafe.Pointer
```

Comment:
IOContext.Pointer() just returns unsafe.Pointer. Both APIs should probably use the same `Pointer()` or `Ptr()` method/type.

Reply:
Agreed, it may be simpler to use unsafe.Pointer here despite the added hint that CPtr gives just because we can't export CPtr outside of go-ceph itself.

Reply:
Well, it was meant to protect against _us_, not some external user. I think we often lost track of that important detail, that's why we discovered some stuff with the GODEBUG setting now. So I don't get the "we can't export" argument. I came up with it, because it helped me getting it sorted while adapting the omap code. But anyway I just pushed another PR an rebased this one on top of it, because I moved it to the existing aliases file. So let's continue there with this topic.